### PR TITLE
Add documentation

### LINKED ISSUE #62: Documentation
## 📚 Documentation
The `nc-time-axis` project needs some basic documentation on readthedocs to help the community understand what the package is and how to use it.

### PR DESCRIPTION
**Pull request by spencerkclark**
## 🚀 Pull Request

Closes #62

### Description

This adds a basic start to the documentation using the sphinx RTD theme.  Just posting this now to make sure I can get the web hooks configured properly.  Will push more substantive updates later.

### CODE REVIEW

**Comment by spencerkclark** [comment]
I think this should now be ready for review.  I added pages for installation instructions, examples, and an API reference.  I also tried to reconstruct release notes dating back to the first release, so that all contributors are acknowledged.  See [here](https://nc-time-axis--87.org.readthedocs.build/en/87/) for a preview of how things look.

Hopefully this serves as a good base that we can iterate more on in future PRs!

**Review by aulemahal — APPROVED** [review]
Nice! Clean and complete but concise documentation, well done! I believe we won't need much more in the future.

Did you add the whole release note page by hand?? That must have taken a long time!

**Comment by trexfeathers** [comment]
Amazing work @spencerkclark! I won't pretend to have looked over the code, but the render looks beautiful 🎁

**Comment by trexfeathers** [comment]
... and I should take @aulemahal's word that this is acceptable and merge without delay!

**Comment by spencerkclark** [comment]
Thanks @aulemahal and @trexfeathers!  Indeed I wrote the release notes by hand -- it wasn't too bad -- I wanted to make sure everyone who laid the foundations of nc-time-axis before it had formal documentation was properly credited :).  If anyone who knows the history better than I do has any corrections or additions I encourage them to make them.

File: docs/installation.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+* `matplotlib <https://matplotlib.org/stable>`_
+* `numpy <https://numpy.org/doc/stable>`_
+* `python <https://docs.python.org/3>`_ >= 3.7
+
+Test requirements
+-----------------
+
+For testing, nc-time-axis also depends on pytest:
+
+* `pytest <https://docs.pytest.org/en/6.2.x>`_ >= 6.0
```
Comment (pelson) [inline docs/installation.rst:19]: It seems surprising to me that you would have the instructions on how to test something before you tell basic users how to install it. (I'm personally in the camp of not bothering to document installation requirements like this, and instead encode it in the ``setup.py`` (or equivalent), but that is not a universal position)

File: docs/release-notes.rst [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+Bug fixes
+~~~~~~~~~
+
+* Fixed a bug the prevented converting NumPy arrays of datetime objects.  This
+  enables making Hovmoller diagrams using nc-time-axis.  By `Peter Killick
+  <https://github.com/DPeterK>`_.
+
+
+v1.0.0 (July 1st, 2016)
```
Comment (pelson) [inline docs/release-notes.rst:172]: This history looks about right. As I recal, it was a prototype from me, and a productionisation from lbdreyer, so think you have the balance of credit about right 👍

File: docs/release-notes.rst [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+-----------------------
+
+New features
+~~~~~~~~~~~~
+
+* Added the fundamental objects of nc-time-axis, i.e.
+  :py:class:`NetCDFTimeDateLocator`, :py:class:`NetCDFTimeDateFormatter`, and
+  :py:class:`NetCDFTimeConverter` (:pull:`2`).  By `lbdryer
+  <https://github.com/lbdreyer>`_ and `Phil Elson <https://github.com/pelson>`_.
+* Added unit and integration tests (:pull:`3`, :pull:`13`).  By `lbdryer
```
Comment (pelson) [inline docs/release-notes.rst:182]: Typo (throughout) of lbdryer -> lbdreyer

**Review by pelson — COMMENTED** [review]
Nicely done 👍 👏

File: docs/installation.rst (hunk 1 above)
Comment (spencerkclark) [reply docs/installation.rst:19]: That makes sense.  I went ahead and removed the testing requirements from the documentation in #89.  It occurred to me there are additional requirements for building the documentation as well, which I had neglected document.  I added those to the `setup.cfg` file where the testing requirements were documented.  Testing and building documentation are both things that I would mainly expect developers to be concerned with, so it is reasonable that they would only be discoverable there.

It is true that having the minimum requirements documented in multiple places opens the door for them to get out of sync, but I do think it is helpful for non-developers to see those without having to look at the source code.

File: docs/release-notes.rst (hunk 2 above)
Comment (spencerkclark) [reply docs/release-notes.rst:172]: Great thanks!  I added a short sentence to this effect in the top-level summary of the initial release in #89, which hopefully makes these contributions by both you and @lbdreyer a little more prominent.

File: docs/release-notes.rst (hunk 3 above)
Comment (spencerkclark) [reply docs/release-notes.rst:182]: Thanks!  Also addressed in #89.

**Comment by spencerkclark** [comment]
I appreciate your comments @pelson!  See #89 for a followup.

**Pull request merged**